### PR TITLE
state(invites): remove `invites-sent` flux store

### PR DESCRIPTION
**Pull request by flootr**
#### Changes proposed in this Pull Request

* Remove the `invites-sent` flux store and replace actions with internal component methods

#### Testing instructions

For some parts of the testing instructions it makes sense to run this branch locally because you have to make changes in code to test a certain error scenario.

To be able to test error responses from the API apply the following diff so you can submit invites despite invalid input.

```diff
diff --git a/client/my-sites/people/invite-people/index.jsx b/client/my-sites/people/invite-people/index.jsx
index f0eaa3f853..acf5e3763e 100644
--- a/client/my-sites/people/invite-people/index.jsx
+++ b/client/my-sites/people/invite-people/index.jsx
@@ -323,6 +323,7 @@ class InvitePeople extends React.Component {
        };
 
        isSubmitDisabled = () => {
+               return false;
                const { success, usernamesOrEmails } = this.state;
                const invitees = Array.isArray( usernamesOrEmails ) ? usernamesOrEmails : [];
```

1. Open this branch on `/people/new/:site`
2. Add valid and invalid usernames and/or emails (test all 3 scenarios: all valid, all invalid and mixed)
3. Hit submit and make sure that valid entries are removed from the form and invalid entries are still there
4. For the last scenario block the request URL via browser devtools, is an error notice shown?

part of #24149

### CODE REVIEW

File: client/my-sites/people/invite-people/index.jsx [hunk 1]
```diff
@@ -245,6 +239,58 @@ class InvitePeople extends React.Component {
...
+						count: usernamesOrEmails.length,
+					} )
+				);
+				this.props.recordTracksEventAction( 'calypso_invite_send_success', { role } );
+			}
+
+			this.refreshFormState( response.errors, response.sent );
+		} catch ( error ) {
+			this.props.errorNotice(
+				this.props.translate( "We couldn't process your invitations. Please try again." )
```
Comment (a8ci18n) [inline client/my-sites/people/invite-people/index.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **21** times:
`translate( 'Sorry, we couldn't process your payment. Please try again later.' )` *ES Score: 6*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/reduxify/invites-sent-new)

File: client/my-sites/people/invite-people/index.jsx [hunk 2]
```diff
@@ -245,6 +239,60 @@ class InvitePeople extends React.Component {
...
+					} )
+				);
+				this.props.recordTracksEventAction( 'calypso_invite_send_success', { role } );
+			}
+
+			this.refreshFormState( response.errors, response.sent );
+		} catch ( error ) {
+			this.props.errorNotice(
+				this.props.translate(
+					"Sorry, we couldn't process your invitations. Please try again later."
```
Comment (a8ci18n) [inline client/my-sites/people/invite-people/index.jsx:286]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **21** times:
`translate( 'Sorry, we couldn't process your payment. Please try again later.' )` *ES Score: 8*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/reduxify/invites-sent-new)

File: client/lib/invites/reducers/invites-sent.js [hunk 3]
```diff
@@ -1,34 +0,0 @@
...
-		case ActionTypes.RECEIVE_SENDING_INVITES_ERROR:
-			return {
-				...state,
-				errors: {
-					...state.errors,
-					[ formId ]: data,
-				},
-			};
-
-		case ActionTypes.RECEIVE_SENDING_INVITES_SUCCESS:
```
Comment (tyxla) [inline client/lib/invites/reducers/invites-sent.js:22]: Seems like we can also delete `RECEIVE_SENDING_INVITES_ERROR` from `lib/invites/constants`.

File: client/lib/invites/reducers/invites-sent.js [hunk 4]
```diff
@@ -1,34 +0,0 @@
...
-import { action as ActionTypes } from 'calypso/lib/invites/constants';
-
-export const initialState = {
-	successes: {},
-	errors: {},
-};
-
-export const reducer = ( state = initialState, { action: { type, data, formId } } ) => {
-	switch ( type ) {
-		case ActionTypes.RECEIVE_SENDING_INVITES_ERROR:
```
Comment (tyxla) [inline client/lib/invites/reducers/invites-sent.js:13]: Seems like we can also delete `RECEIVE_SENDING_INVITES_SUCCESS` from `lib/invites/constants`.

File: client/lib/invites/actions.js [hunk 5]
```diff
@@ -123,72 +122,3 @@ export function acceptInvite( invite, callback ) {
 		} );
 	};
 }
-
-export function sendInvites( siteId, usernamesOrEmails, role, message, formId, isExternal ) {
-	return ( dispatch ) => {
-		Dispatcher.handleViewAction( {
-			type: ActionTypes.SENDING_INVITES,
```
Comment (tyxla) [inline client/lib/invites/actions.js:130]: Seems like we can also delete `SENDING_INVITES` from `lib/invites/constants`.

File: client/my-sites/people/invite-people/index.jsx [hunk 6]
```diff
@@ -110,16 +102,15 @@ class InvitePeople extends React.Component {
...
 			this.setState( this.resetState() );
 			this.props.recordTracksEventAction( 'calypso_invite_people_form_refresh_initial' );
 			debug( 'Submit successful. Resetting form.' );
-		} else {
-			const sendInvitesErrored = InvitesSentStore.getErrors( this.state.formId );
-			const errors = get( sendInvitesErrored, 'errors', {} );
+			return;
+		}
+
+		if ( Object.keys( errors ).length ) {
```
Comment (tyxla) [inline client/my-sites/people/invite-people/index.jsx]: This is the second time we're doing a `Object.keys( errors )` in this method. Should we move it to a `const`?

File: client/my-sites/people/invite-people/index.jsx [hunk 7]
```diff
@@ -728,10 +768,11 @@ const connectComponent = connect(
 	( dispatch ) => ( {
 		...bindActionCreators(
```
Comment (tyxla) [inline client/my-sites/people/invite-people/index.jsx]: Optional: we can totally get rid of this one - binding actions to `dispatch` is now automatic when we return an object with all unbound action creators. Wanna remove this thorn out of my eye? 😉

**Review by tyxla — APPROVED** [review]
This is testing well and code looks good 👍 

I've left some suggestions, nothing blocking.

Nice work 👏

**Pull request merged**